code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -662,17 +662,17 @@
 			continue
 
 		//compute volume & ph like we would normally
-		. += reagent_volume
+		running_volume += reagent_volume
 		total_ph += (reagent.ph * reagent_volume)
 
 		//reasign rounded value
 		reagent.volume = reagent_volume
 
 	//assign the final values, rounding up can sometimes cause overflow so bring it down
-	total_volume = min(round(., CHEMICAL_VOLUME_ROUNDING), maximum_volume)
-	if(!.)
+	total_volume = min(round(running_volume, CHEMICAL_VOLUME_ROUNDING), maximum_volume)
+	if(!running_volume)
 		ph = CHEMICAL_NORMAL_PH
-	else
+	else if(total_volume != 0)
```

Comment:
```suggestion
	else if(total_volume)
```

Why is the ph based on the rounded volume though?

Reply:
Not too sure, the original code had it like that so I didn't want to mess with it too much, but it's probably not necessary.

Reply:
The volume had to be rounded so that values like 49.999 become 50. Why we divide total ph by the total volume that even i don't know. That has been the case for years so i would say it is nessassary and would not touch it